trieste/models/gpflux/models.py

```diff
@@ -0,0 +1,158 @@
+        :param optimizer: The optimizer with which to train the model. Defaults to
+            :class:`~trieste.models.optimizer.TFOptimizer` with :class:`~tf.optimizers.Adam`. Only
+            the optimizer itself is used; other args relevant for fitting should be passed as part
+            of `fit_args`.
+        :param fit_args: A dictionary of arguments to be used in the Keras `fit` method. Default to
+            using 100 epochs, batch size 100, and verbose 0.
+        """
+
+        super().__init__()
+
+        if optimizer is None:
+            self._optimizer = tf.optimizers.Adam()
+        else:
+            self._optimizer = optimizer
+
+        if not isinstance(self._optimizer, tf.keras.optimizers.Optimizer):
+            raise ValueError("Must use a Keras/TF optimizer for DGPs, not wrapped in TFOptimizer")
+
+        self.original_lr = self._optimizer.lr.numpy()
+
+        if fit_args is None:
+            self.fit_args = dict(
+                {
+                    "verbose": 0,
+                    "epochs": 100,
+                    "batch_size": 100,
+                }
+            )
+        else:
+            self.fit_args = fit_args
+
+        if not all([isinstance(layer, (GPLayer, LatentVariableLayer)) for layer in model.f_layers]):
+            raise ValueError(
+                "`DeepGaussianProcess` can only be built out of `GPLayer` or"
+                "`LatentVariableLayer`"
+            )
+
+        self._model_gpflux = model
+
+        self._model_keras = model.as_training_model()
+        self._model_keras.compile(self._optimizer)
+
+    def __repr__(self) -> str:
+        """"""
+        return f"DeepGaussianProcess({self._model_gpflux!r}, {self.optimizer!r})"
+
+    @property
+    def model_gpflux(self) -> DeepGP:
+        return self._model_gpflux
+
+    @property
+    def model_keras(self) -> tf.keras.Model:
+        return self._model_keras
+
+    @property
+    def optimizer(self) -> tf.keras.optimizers.Optimizer:
+        return self._optimizer
+
+    def sample(self, query_points: TensorType, num_samples: int) -> TensorType:
+        def get_samples(query_points: TensorType, num_samples: int) -> TensorType:
+            samples = []
+            for _ in range(num_samples):
+                samples.append(sample_dgp(self.model_gpflux)(query_points))
+            return tf.stack(samples)
+
+        return get_samples(query_points, num_samples)
+
+    def update(self, dataset: Dataset) -> None:
+        inputs = dataset.query_points
+        new_num_data = inputs.shape[0]
+        self.model_gpflux.num_data = new_num_data
+
+        # Update num_data for each layer, as well as make sure dataset shapes are ok
+        for i, layer in enumerate(self.model_gpflux.f_layers):
+            layer.num_data = new_num_data
+
+            if isinstance(layer, LatentVariableLayer):
+                inputs = layer(inputs)
+                continue
```

Comment:
why do you need `continue` here?

Reply:
`LatentVariableLayer`s don't have inducing points, so we just propagate the inputs and skip.